racksdb/web/ui/src/views/DatacenterRoomView.vue

```diff
@@ -0,0 +1,106 @@
+<script setup lang="ts">
+import { useHttp } from '@/plugins/http';
+import { ref, onMounted, inject } from 'vue'
+import SearchBar from '@/components/SearchBar.vue';
+import { injectionKey } from '@/plugins/runtimeConfiguration';
+import type { Ref } from 'vue'
+import type { Datacenter } from './DatacentersView.vue';
+
+const datacenters: Ref<Array<Datacenter>> = ref([])
+const datacenterDetails: Ref<Datacenter | undefined> = ref()
+const racks: Ref<Array<Rack>> = ref([])
+const rackDetails: Ref<Array<Rack>> = ref([])
+const showFullImg = ref(false)
+const http = useHttp()
+
+function openImg(){
+    showFullImg.value = !showFullImg.value
+}
+
+function listInfrastructures(){
```

Comment:
Please add comment to explain what this function is supposed to do and what it returns.

Considering where this function is called, you want to list all infrastructures in a specific rack. Currently, it returns the infrastructure of the first node in the first rack. The for loop is not even used as it returns on the 1st iteration. It must be fixed.